# Forecast of Sinlge Family Homes

In [6]:
#importing libraries

In [7]:
import pandas as pd
import os

In [8]:
data_frame=pd.read_csv('Monthly_Sale_Price_of_Single_Family_Homes_in_CT.csv')

In [9]:
data_frame

Date  Median Sale Price  Average Sale Price     County
0     01/01/2001 12:00:00 AM          369587.11           607031.57  Fairfield
1     02/01/2001 12:00:00 AM          351716.13           612695.88  Fairfield
2     03/01/2001 12:00:00 AM          375670.85           581427.50  Fairfield
3     04/01/2001 12:00:00 AM          372628.98           579640.90  Fairfield
4     05/01/2001 12:00:00 AM          401488.72           576070.59  Fairfield
...                      ...                ...                 ...        ...
2245  06/01/2021 12:00:00 AM          245500.00           255984.24    Windham
2246  07/01/2021 12:00:00 AM          259950.00           273981.86    Windham
2247  08/01/2021 12:00:00 AM          250000.00           263424.27    Windham
2248  09/01/2021 12:00:00 AM          250000.00           272642.60    Windham
2249  10/01/2021 12:00:00 AM          250000.00           265434.49    Windham

[2250 rows x 4 columns]

In [10]:
data_frame['County'].value_counts()

Fairfield      250
Hartford       250
Litchfield     250
Middlesex      250
New Haven      250
New London     250
Connecticut    250
Tolland        250
Windham        250
Name: County, dtype: int64

Each county has a equal share of 250 values of median and average sale prices

In [11]:
#Extracting only date from Date Feature
dates=[]
for val in data_frame['Date']:
    dates.append(val[:10])

In [12]:
data_frame['Date']=dates

In [13]:
data_frame['Date']=pd.to_datetime(data_frame['Date'],format='%m/%d/%Y')

In [14]:
data_frame.head()

Date  Median Sale Price  Average Sale Price     County
0 2001-01-01          369587.11           607031.57  Fairfield
1 2001-02-01          351716.13           612695.88  Fairfield
2 2001-03-01          375670.85           581427.50  Fairfield
3 2001-04-01          372628.98           579640.90  Fairfield
4 2001-05-01          401488.72           576070.59  Fairfield

In [15]:
print(pd.DatetimeIndex(data_frame['Date']).month)
print(pd.DatetimeIndex(data_frame['Date']).day)
print(pd.DatetimeIndex(data_frame['Date']).year)

Int64Index([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10,
            ...
             1,  2,  3,  4,  5,  6,  7,  8,  9, 10],
           dtype='int64', name='Date', length=2250)
Int64Index([1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
            ...
            1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
           dtype='int64', name='Date', length=2250)
Int64Index([2001, 2001, 2001, 2001, 2001, 2001, 2001, 2001, 2001, 2001,
            ...
            2021, 2021, 2021, 2021, 2021, 2021, 2021, 2021, 2021, 2021],
           dtype='int64', name='Date', length=2250)


Extracting each dataframe based on a specific County

In [16]:
fairfield_data=data_frame[data_frame['County']=='Fairfield']
hartford_data=data_frame[data_frame['County']=='Hartford']
litchfield_data=data_frame[data_frame['County']=='Litchfield']
middlesex_data=data_frame[data_frame['County']=='Middlesex']
new_haven_data=data_frame[data_frame['County']=='New Haven']
new_london_data=data_frame[data_frame['County']=='New London']
connecticut_data=data_frame[data_frame['County']=='Connecticut']
tolland_data=data_frame[data_frame['County']=='Tolland']
windham_data=data_frame[data_frame['County']=='Windham']

In [17]:
#assiging the dataframes into list

In [18]:
data_list=[fairfield_data,hartford_data,litchfield_data,middlesex_data,new_haven_data,
           new_london_data,connecticut_data,tolland_data,windham_data]

In [19]:
#extracting the county names

In [20]:
data_names=data_frame['County'].value_counts().index

In [21]:
#each county data frame is saved into new separate folders named with county name

In [22]:
parent_dir='/Users/tirumaleshn2000/Projects to modify/House prices forecast/'
mode = 0o666
for data,name in zip(data_list,data_names):
    directory=name
    path=parent_dir+directory
    file_name=path+'/'+name+'.csv'
    try:
        #if the folder not exists, new folder is created in the parent directory.
        os.mkdir(path)
        print("Directory '% s' created" % directory)
        data.to_csv(file_name,index=False)
        print('Dataframe {} saved'.format(name))
    except:
        #exception is raised if the folder already exists
        print('Folder already exists')

Directory 'Fairfield' created
Dataframe Fairfield saved
Directory 'Hartford' created
Dataframe Hartford saved
Directory 'Litchfield' created
Dataframe Litchfield saved
Directory 'Middlesex' created
Dataframe Middlesex saved
Directory 'New Haven' created
Dataframe New Haven saved
Directory 'New London' created
Dataframe New London saved
Directory 'Connecticut' created
Dataframe Connecticut saved
Directory 'Tolland' created
Dataframe Tolland saved
Directory 'Windham' created
Dataframe Windham saved
